api/src/test/java/com/epam/pipeline/controller/log/LogControllerTest.java

```diff
@@ -0,0 +1,119 @@
+
+@WebMvcTest(controllers = LogController.class)
+public class LogControllerTest extends AbstractControllerTest {
+
+    @Autowired
+    private LogApiService mockLogApiService;
+
+    private static final String LOG_ENDPOINT = SERVLET_PATH + "/log/filter";
+    private final LogFilter logFilter = new LogFilter();
+
+    @Test
+    public void shouldFailForUnauthorizedUserGet() throws Exception {
+        mvc().perform(get(LOG_ENDPOINT)
+                .servletPath(SERVLET_PATH))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnLogFilter() throws Exception {
+        logFilter.setSortOrder("ASC");
+        Mockito.doReturn(logFilter).when(mockLogApiService).getFilters();
+        final MvcResult mvcResult = mvc().perform(get(LOG_ENDPOINT)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE))
+                .andExpect(status().isOk()).andReturn();
+        Mockito.verify(mockLogApiService).getFilters();
+
+        final ResponseResult<LogFilter> expectedResult = new ResponseResult<>();
+        expectedResult.setStatus("OK");
+        expectedResult.setPayload(logFilter);
+
+        final String actual= mvcResult.getResponse().getContentAsString();
+        assertThat(actual).isEqualToIgnoringWhitespace(getObjectMapper().writeValueAsString(expectedResult));
+
+        final Result<LogFilter> actualResult =
+                JsonMapper.parseData(actual, new TypeReference<Result<LogFilter>>() { });
+
+        assertEquals(expectedResult.getPayload(), actualResult.getPayload());
+    }
+
+    @Test
+    public void shouldFailForUnauthorizedUserPost() throws Exception {
+        mvc().perform(post(LOG_ENDPOINT)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE)
+                .content(getObjectMapper().writeValueAsString(logFilter)))
+                .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldReturnFilteredLogs() throws Exception {
+        final LogPagination logPagination = LogPagination.builder().pageSize(5).build();
+        logFilter.setMessage("testMessage");
+        Mockito.doReturn(logPagination).when(mockLogApiService).filter(logFilter);
+        final MvcResult mvcResult = mvc().perform(post(LOG_ENDPOINT)
+                .servletPath(SERVLET_PATH)
+                .contentType(EXPECTED_CONTENT_TYPE)
+                .content(getObjectMapper().writeValueAsString(logFilter)))
+                .andExpect(status().isOk()).andReturn();
+
+        final ArgumentCaptor<LogFilter> logFilterCaptor = ArgumentCaptor.forClass(LogFilter.class);
+        Mockito.verify(mockLogApiService).filter(logFilterCaptor.capture());
+        assertThat(logFilterCaptor.getValue().getMessage()).isEqualTo("testMessage");
+
+        final ResponseResult<LogPagination> expectedResult = new ResponseResult<>();
```

Comment:
Could we move this logic into separate generic method? (to use it in other test)

smth like:
```
private <T> ResponseResult<T> buildExpectedResult(final T payload) {
        final ResponseResult<T> expectedResult = new ResponseResult<>();
        expectedResult.setStatus("OK");
        expectedResult.setPayload(payload);
        return expectedResult;
    }
```

P.S.: may be should create additional utility class for controller tests